src/wobjectimpl.h

```diff
@@ -841,6 +846,10 @@ struct FriendHelper {
         return _id;
     }
 
+#ifdef Q_CC_GNU
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress"
```

Comment:
Qt provides macro for this:

QT_WARNING_PUSH QT_WARNING_DISABLE_GCC QT_WARNING_POP

Reply:
I did never use those… added them